terraform-ss-proxy/install.sh

```diff
@@ -0,0 +1,217 @@
+
+    # Prepare ratelimit path
+    sudo mkdir -p /var/lib/apache2/ratelimit
+    sudo chown www-data:www-data /var/lib/apache2/ratelimit
+
+    # Add Config
+    sudo cp unity-cs-main.conf /etc/apache2/sites-enabled/unity-cs-main.conf
+
+    # Update with system parameters
+    # First lookup the client secret
+    echo "Looking up Cognito client secret..."
+    CLIENT_SECRET=$(aws cognito-idp describe-user-pool-client \
+      --user-pool-id ${COGNITO_USER_POOL_ID} \
+      --client-id ${OIDC_CLIENT_ID} \
+      --region ${AWS_REGION} \
+      --query 'UserPoolClient.ClientSecret' \
+      --output text)
+    
+    if [ -z "$CLIENT_SECRET" ]; then
+      echo "Error: Could not retrieve Cognito client secret. Check your credentials and permissions."
+      exit 1
+    else
+      echo "Successfully retrieved Cognito client secret."
+    fi
+    
+    # Update the configuration with the pool ID and client secret
+    sudo sed -i "s/\${COGNITO_POOL_ID}/${COGNITO_USER_POOL_ID}/" /etc/apache2/sites-enabled/unity-cs-main.conf
+    sudo sed -i "s/\${OIDC_CLIENT_ID}/${OIDC_CLIENT_ID}/" /etc/apache2/sites-enabled/unity-cs-main.conf
+    sudo sed -i "s/\${OIDC_CLIENT_SECRET}/${CLIENT_SECRET}/" /etc/apache2/sites-enabled/unity-cs-main.conf
+    sudo sed -i "s/\${PORT_NUM}/${APACHE_PORT}/" /etc/apache2/sites-enabled/unity-cs-main.conf
+
+    # Remove the default
+    sudo rm /etc/apache2/sites-enabled/000-default.conf
+
+    # Generate a cert
+    # Generate certificate and key with predefined values to avoid prompts
+    sudo mkdir -p /etc/ssl/certs/
+    sudo mkdir -p /etc/ssl/private/
+    sudo openssl req -x509 -nodes -days 3650 -newkey rsa:2048 \
+        -keyout /etc/ssl/private/apache-selfsigned.key \
+        -out /etc/ssl/certs/apache-selfsigned.crt \
+        -subj "/C=US/ST=CA/L=Pasadena/O=MDPS/OU=IT/CN=localhost"
+
+    # Set proper permissions
+    echo "Setting permissions..."
+    sudo chmod 600 /etc/ssl/private/apache-selfsigned.key
+    sudo chmod 644 /etc/ssl/certs/apache-selfsigned.crt
+
+    # Create remove default index.html
+    sudo rm /var/www/html/index.html
+
+    # Prepare CGI Script
+    sudo mkdir -p /etc/apache2/venues.d/
+    sudo chown www-data:www-data /etc/apache2/venues.d/
+    sudo cp reload-apache.cgi /usr/lib/cgi-bin/reload-apache.cgi
+    
+    # Replace S3 bucket placeholder in CGI script
+    sudo sed -i "s#REPLACE_WITH_S3_BUCKET_NAME#${S3_BUCKET_NAME}#g" /usr/lib/cgi-bin/reload-apache.cgi
+    
+    # Replace environment placeholder in CGI script
+    sudo sed -i "s#REPLACE_WITH_ENVIRONMENT_NAME#${ENVIRONMENT}#g" /usr/lib/cgi-bin/reload-apache.cgi
+    
+    sudo chown www-data:www-data /usr/lib/cgi-bin/reload-apache.cgi
+    sudo chmod 755 /usr/lib/cgi-bin/reload-apache.cgi
+
+    # Update sudoers
+    # Using tee to append to sudoers to handle permissions, and ensuring lines are added only once.
+    if ! sudo grep -q "www-data ALL=(ALL) NOPASSWD: /usr/sbin/apachectl configtest" /etc/sudoers; then
```

Comment:
Just a nitpick, but everywhere else you've been good to wrap IF conditions with brackets- why not here? also, Probably better to compare the actual return code `-ne 0` or something.